addons/account/models/account_invoice.py

```diff
@@ -1272,6 +1274,12 @@ def _set_taxes(self):
         else:
             self.price_unit = fix_price(self.product_id.lst_price, taxes, fp_taxes)
 
+    @api.onchange('price_subtotal')
+    def _onchange_price_subtotal(self):
+        for line in self:
```

Comment:
one line can be negative (discount line for example) but the total of all lines cannot.

So you shouldn't raise the warning if the invoice total doesn't fall bellow 0 after this line.